components/src/components/elements/checkout-dialog/CheckoutDialog.tsx

```diff
@@ -273,14 +273,21 @@ export const CheckoutDialog = () => {
     };
   }, []);
 
-  const allowCheckout =
+  const allowSubmit =
```

Comment:
Open to feedback on these names. I think you mentioned as well that this is likely to change as we add more stages (eg for add-ons0

Reply:
How about `hasValidPlan` and `canCheckout`?

```js
const hasValidPlan =
  api &&
  selectedPlan &&
  (selectedPlan.id !== currentPlan?.id ||
    planPeriod !== currentPlan.planPeriod) &&
  !isLoading;

const canCheckout =
  hasValidPlan && ((paymentMethod && !showPaymentForm) || paymentMethodId);
```

Reply:
Wait, no! `canUpdateSubscription`.